ext/opentelemetry-ext-wsgi/src/opentelemetry/ext/wsgi/__init__.py

```diff
@@ -88,14 +118,27 @@ def __call__(self, environ, start_response):
         tracer = trace.tracer()
         path_info = environ["PATH_INFO"] or "/"
 
-        with tracer.start_span(path_info, kind=trace.SpanKind.SERVER) as span:
+        span = tracer.create_span(path_info, kind=trace.SpanKind.SERVER)
+        span.start()
+        try:
             self._add_request_attributes(span, environ)
             start_response = self._create_start_response(span, start_response)
 
             iterable = self.wsgi(environ, start_response)
-            try:
-                for yielded in iterable:
-                    yield yielded
-            finally:
-                if hasattr(iterable, "close"):
-                    iterable.close()
+
+            # Put this in a subfunction to not delay the call to the wrapped
+            # WSGI application (instrumentation should change the application
+            # behavior as little as possible).
+            def iter_result():
+                try:
+                    for yielded in iterable:
+                        yield yielded
+                finally:
+                    if hasattr(iterable, "close"):
+                        iterable.close()
+                    span.end()
+
+            return iter_result()
```

Comment:
I don't think this changes behavior but this return might be better off outside of the try/except to make the behavior more clear?

Reply:
If this fails for some reason (e.g. KeyboardInterrupt), I'd still like to end the Span.

Reply:
See also https://github.com/open-telemetry/opentelemetry-python/pull/148#discussion_r325980107, let's gather discussion there.